src/main/java/org/alfresco/repo/web/scripts/RepositoryContainer.java

```diff
@@ -515,7 +515,8 @@ public Object execute() throws Exception
                             {
                                 // Reset the request and response in case of a transaction retry
                                 bufferedReq.reset();
-                                bufferedRes.reset();
+                                // REPO-4388 don't reset CORS headers if exist
+                                bufferedRes.reset("Access-Control-.*");

```

Comment:
We should externalize this, set at as a bean property, and allow it to be overwritten